src/App.tsx

```diff
@@ -15,68 +15,93 @@ export const goodsFromServer = [
+
+    setGoods(sortedGoods);
+    setSortType('alphabetical');
+    setIsReversed(false);
+  };
+
+  const sortByLength = () => {
+    const sortedGoods = [...goodsFromServer].sort(
+      (a, b) => a.length - b.length,
+    );
 
-// Use this function in the render method to prepare goods
-export function getReorderedGoods(
-  goods: string[],
-  { sortType, isReversed }: ReorderOptions,
-) {
-  // To avoid the original array mutation
-  const visibleGoods = [...goods];
+    setGoods(sortedGoods);
+    setSortType('length');
+    setIsReversed(false);
+  };
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  const reverseGoods = () => {
+    const reversedGoods = [...goods].reverse();
 
-  return visibleGoods;
-}
+    setGoods(reversedGoods);
+    setIsReversed(!isReversed);
+  };
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
+  const resetGoods = () => {
+    setGoods([...goodsFromServer]);
+    setSortType('');
+    setIsReversed(false);
+  };
+
+  const isOriginalOrder =
+    JSON.stringify(goods) === JSON.stringify(goodsFromServer);
 
-export const App: React.FC = () => {
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={`button is-info ${
+            sortType === 'alphabetical' && !isReversed ? '' : 'is-light'
+          }`}
+          onClick={sortAlphabetically}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={`button is-success ${
+            sortType === 'length' && !isReversed ? '' : 'is-light'
+          }`}
+          onClick={sortByLength}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={`button is-warning ${isReversed ? '' : 'is-light'}`}
+          onClick={reverseGoods}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {!isOriginalOrder && (
+          <button
+            type="button"
+            className="button is-danger"
+            onClick={resetGoods}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {goods.map((good, index) => (
+          <li key={index} data-cy="Good">
```

Comment:
It's a bad practice to use index as key
```suggestion
        {goods.map((good) => (
          <li key={good} data-cy="Good">
```